### PR TITLE
pass std::string_view by value

### LINKED ISSUE #5404: Pass std::string_view by value
## Refactoring request

Refine some functions to pass std::string_view by value:
https://github.com/StarRocks/starrocks/search?q=const+std%3A%3Astring_view%26

Reference:
https://quuxplusone.github.io/blog/2021/11/09/pass-string-view-by-value/

### PR DESCRIPTION
**Pull request by zombee0**
Signed-off-by: zombee0 <flylucas_10@163.com>

## What type of PR is this：
- [ ] enhancement

## Which issues of this PR fixes ：

Fixes #5404 

## Problem Summary(Required) ：

pass the std::string_view by value, because there are at least three performance benefits, 1.Pass-by-value eliminates a pointer indirection in the callee; 2.Pass-by-ref need address, and Pass-by-value eliminates the need to spill the argumen,such as (byvalue("hello") vs byref("hello")); 3.Eliminate aliasing,Pass-by-ref, the callee know nothing about the ref, maybe some other pointers point to the object, and change the object, while Pass-by-value, the callee has a copy.

### CODE REVIEW

File: be/src/util/timezone_utils.h [hunk 1]
```diff
@@ -34,11 +34,11 @@ namespace starrocks {
 
 class TimezoneUtils {
 public:
-    static bool find_cctz_time_zone(const TimezoneHsScan& timezone_hsscan, const std::string_view& timezone,
+    static bool find_cctz_time_zone(const TimezoneHsScan& timezone_hsscan, const std::string_view timezone,
                                     cctz::time_zone& ctz);
     static void init_time_zones();
-    static bool find_cctz_time_zone(const std::string_view& timezone, cctz::time_zone& ctz);
-    static bool timezone_offsets(const std::string_view& src, const std::string_view& dst, int64_t* offset);
```
Comment (imay) [inline be/src/util/timezone_utils.h:41]: `const std::string_view&` -> `std::string_view`

**Comment by imay** [comment]
@zombee0 If `starrocks_clang-formaat` don't pass, you can run `./build-support/clang-format.sh` to make code formatted.

**Comment by zombee0** [comment]
@imay i had run ./build-support/clang-format.sh on my computer, and there it no error, but still don't pass

**Comment by imay** [comment]
what's the `clang-format --version` output. 
It works only for version 10.0.*

**Comment by zombee0** [comment]
@imay thank you so much, i will retry!

**Review by imay — APPROVED** [review]
LGTM

**Review by sduzh — APPROVED** [review]

**Pull request merged**